### PR TITLE
cadl, flatten payload, emitter

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
link https://github.com/Azure/autorest.java/issues/1898

case https://github.com/Azure/autorest.java/blob/9d916ded53c2af25deacdd1f4ed50646312a425c/cadl-tests/cadl/flatten.cadl#L33-L36

### CODE REVIEW

File: cadl-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -624,6 +625,56 @@ export class CodeModelBuilder {
...
+                  },
+                ),
+              );
+            }
+          }
+        }
+      }
+
+      // TODO remove the line after full support on anonymous model
+      op.convenienceApi = undefined;
```
Comment (weidongxu-microsoft) [inline cadl-extension/src/code-model-builder.ts:676]: convenience API disable before full support.

protocol API is good.

**Comment by weidongxu-microsoft** [comment]
javagen part, we likely reuse `MethodTransformationDetail` for client model in client method.

However, this structure might not be good for sample code generate.

File: cadl-extension/src/code-model-builder.ts (hunk 1 above)
Comment (haolingdong-msft) [reply cadl-extension/src/code-model-builder.ts:676]: ok, so the alias name will not be used? in our case, the name `Request`.
```
 alias Request = { 
   user?: User; 
   input: string; 
 }; 
```

Comment (weidongxu-microsoft) [reply cadl-extension/src/code-model-builder.ts:676]: alias is only exist when writing Cadl source.

It is not visible from emitter.

**Pull request merged**